hummingbot/core/utils/gateway_http_client.py

```diff
@@ -0,0 +1,83 @@
+import aiohttp
+from typing import Dict, Any, Optional
+import ssl
+import json
+from hummingbot.client.config.global_config_map import global_config_map
+from hummingbot.client.settings import GATEAWAY_CA_CERT_PATH, GATEAWAY_CLIENT_CERT_PATH, GATEAWAY_CLIENT_KEY_PATH
+from hummingbot.logger import HummingbotLogger
+import logging
+
+
+class GatewayHttpClient:
+    """
+    An HTTP client for making requests to the gateway API.
+    """
+
+    _ghc_logger: Optional[HummingbotLogger] = None
+    _shared_client: Optional[aiohttp.ClientSession] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._ghc_logger is None:
+            cls._ghc_logger = logging.getLogger(__name__)
+        return cls._ghc_logger
+
+    def __init__(self):
+        self._http_client()
+
+    def _http_client(self) -> aiohttp.ClientSession:
+        """
+        :returns Shared client session instance
+        """
+        if self._shared_client is None:
+            ssl_ctx = ssl.create_default_context(cafile=GATEAWAY_CA_CERT_PATH)
+            ssl_ctx.load_cert_chain(GATEAWAY_CLIENT_CERT_PATH, GATEAWAY_CLIENT_KEY_PATH)
+            conn = aiohttp.TCPConnector(ssl_context=ssl_ctx)
+            self._shared_client = aiohttp.ClientSession(connector=conn)
```

Comment:
`ClientSession` should not be initialized outside of asyncio run loop - the way you're initializing it right now is improper.

You should use late binding instead. Do NOT initialize the shared http client in `__init__()`. Instead, turn this function into a property and do late binding initialization when it's called by `api_request()`. Since `api_request()` is an async function, you're guaranteed to be in a run loop when your `http_client` property gets called.

Reply:
I made changes. Is it what you're expecting?